sdk/cosmos/azure-cosmos/azure/cosmos/_base.py

```diff
@@ -215,7 +215,12 @@ def GetHeaders(  # pylint: disable=too-many-statements,too-many-branches
             headers[http_constants.HttpHeaders.PartitionKey] = []
         # else serialize using json dumps method which apart from regular values will serialize None into null
         else:
-            headers[http_constants.HttpHeaders.PartitionKey] = json.dumps([options["partitionKey"]])
+            # single partitioning uses a string and needs to be turned into a list
+            if isinstance(options["partitionKey"], list) and options["partitionKey"]:
```

Comment:
We already do the check for partitionKey in options on line 209
```suggestion
            if isinstance(options["partitionKey"], list):
```

Reply:
In line 209 we are checking if options was given a partition key, but it does not check if that partition key is None or [] in which case we need this check to avoid an edge case that would result in an error.